client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/CacheBasedDeviceConnectionInfo.java

```diff
@@ -167,39 +164,26 @@ public HotrodBasedDeviceConnectionInfo(final RemoteCache<String, String> cache,
         Objects.requireNonNull(adapterInstanceId);
 
         final String key = getAdapterInstanceEntryKey(tenantId, deviceId);
-        return cache.getWithVersion(key)
-               .recover(t -> {
+
+        return cache
+                .remove(key, adapterInstanceId)
+                .recover(t -> {
                     LOG.debug("failed to get cache entry when trying to remove command handling adapter instance [tenant: {}, device-id: {}, adapter-instance: {}]",
```

Comment:
The content of the log message doesn't fit here any more.

Reply:
Fixed.